include/highfive/bits/H5Easy_Eigen.hpp

```diff
@@ -0,0 +1,153 @@
+/*
+ *  Copyright (c), 2017, Adrien Devresse <adrien.devresse@epfl.ch>
+ *
+ *  Distributed under the Boost Software License, Version 1.0.
+ *    (See accompanying file LICENSE_1_0.txt or copy at
+ *          http://www.boost.org/LICENSE_1_0.txt)
+ *
+ */
+#ifndef H5EASY_BITS_EIGEN_HPP
+#define H5EASY_BITS_EIGEN_HPP
+
+#include "../H5Easy.hpp"
+#include "H5Easy_misc.hpp"
+#include "H5Easy_scalar.hpp"  // to get the basic "load_impl"
+
+#ifdef HIGHFIVE_EIGEN
+
+namespace HighFive {
+
+namespace detail {
+
+namespace eigen {
+
+// return the shape of the "Eigen::Matrix" as size 1 or 2 "std::vector<size_t>"
+template <class C, int Rows, int Cols, int Options, int MaxRows, int MaxCols>
+std::vector<size_t> shape(const Eigen::Matrix<C,Rows,Cols,Options,MaxRows,MaxCols>& data)
+{
+    if (Rows == 1) {
+        return {static_cast<size_t>(data.cols())};
+    } else if (Cols == 1) {
+        return {static_cast<size_t>(data.rows())};
+    } else {
+        return {static_cast<size_t>(data.rows()), static_cast<size_t>(data.cols())};
+    }
+}
+
+// write to open DataSet of the correct size
+// if the "Eigen::Matrix" is column-major reordering has to be done first; HDF5 is always row-major
+template <class C, int Rows, int Cols, int Options, int MaxRows, int MaxCols>
+void write(HighFive::DataSet& dataset, const Eigen::Matrix<C,Rows,Cols,Options,MaxRows,MaxCols>& data)
+{
+    if (data.IsRowMajor) {
+        dataset.write(data.data());
+    } else {
+        Eigen::Matrix<C, Rows, Cols, Eigen::RowMajor, MaxRows, MaxCols> tmp = data;
+        detail::eigen::write(dataset, tmp);
+    }
+}
+
+// create DataSet and write data
+template <class T>
+struct dump_impl
+{
+    template <class C, int Rows, int Cols, int Options, int MaxRows, int MaxCols>
+    static HighFive::DataSet run(HighFive::File& file, const std::string& path,
+        const Eigen::Matrix<C,Rows,Cols,Options,MaxRows,MaxCols>& data)
```

Comment:
Format: args just got same indentation as body. inconsistent and puzzling. Please keep the default.